src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -3229,10 +3231,25 @@ ConnectionStateMachine(WorkerSession *session)
 					 * Similarly when the pool is failed over to local execution, warning
 					 * the user just creates chatter.
 					 */
+					IncrementStatCounter(STAT_CONNECTION_OPTIONAL_SKIPPED);
```

Comment:
This could be a separate bucket like "STAT_CONNECTION_FAILED_BUT_OK" but I don't want to complicate the things from the users' perspective. Plus, in a sense, connections to local node can be thought as optional connections too.